PercolationStats.java

```diff
@@ -1,13 +1,95 @@
-import edu.princeton.cs.algs4.StdRandom;
-import edu.princeton.cs.algs4.StdStats;
-import edu.princeton.cs.algs4.WeightedQuickUnionUF;
 
 public class PercolationStats {
-   public PercolationStats(int n, int trials)    // perform trials independent experiments on an n-by-n grid
-   public double mean()                          // sample mean of percolation threshold
-   public double stddev()                        // sample standard deviation of percolation threshold
-   public double confidenceLo()                  // low  endpoint of 95% confidence interval
-   public double confidenceHi()                  // high endpoint of 95% confidence interval
-
-   public static void main(String[] args)        // test client (described below)
-}
\ No newline at end of file
+	private int gridDimension; // number of columns/rows in N x N grid
+	private int numberOfExperiment; // number of times to run stats
+	private double[] results; // stores results of percolation experiments
+	private boolean print_grid = false;
+
+	/**
+	 * Runs T percolation experiments on an N x N grid, and collects the results
+	 * (percolation threshold of each experiment)
+	 * 
+	 * @param N
+	 *            the number of nodes in a single row/column of an NxN grid
+	 * @param T
+	 *            the number of percolation experiments to run
+	 */
+	public PercolationStats(int N, int T) {
+		if (N <= 0 || T <= 0)
+			throw new IllegalArgumentException("N&T cannot be less than 1 ");
+		// loop through numberOfExperiment times
+		gridDimension = N;
+		numberOfExperiment = T;
+		double total_blocks = gridDimension * gridDimension;
+		results = new double[numberOfExperiment];
+		for (int x = 0; x < numberOfExperiment; x++) {
+			Percolation sample = new Percolation(gridDimension);
+			int open_count = 0;
```

Comment:
This variable would have not been needed if you had implemented numberOfOpenSites()